src/main/scala/io/pravega/connectors/spark/PravegaSourceProvider.scala

```diff
@@ -1,519 +1,570 @@
+            new TransactionPravegaWriter(
+              scopeName,
+              streamName,
+              clientConfig,
+              transactionTimeoutMs,
+              readAfterWriteConsistency,
+              transactionStatusPollIntervalMs,
+              inputSchema)
+          } else {
+            new NonTransactionPravegaWriter(
+              scopeName,
+              streamName,
+              clientConfig,
+              inputSchema)
+          }
+        }
+
+
+
+        override def truncate(): WriteBuilder = this
+        override def update(): WriteBuilder = this
+      }
+    }
+  }
+
+  class PravegaMetaScan(options: CaseInsensitiveStringMap, metadataTableName: MetadataTableName.MetadataTableName) extends Scan {
+    log.info("in PravegaMetaScan")
+    override def readSchema(): StructType = {
+      metadataTableName match {
+          case MetadataTableName.StreamInfo => {
+            StructType(Seq(
+              StructField("head_stream_cut", StringType),
+              StructField("tail_stream_cut", StringType)
+            ))
+          }
+          case MetadataTableName.Streams => {
+            StructType(Seq(
+              StructField("scope_name", StringType),
+              StructField("stream_name", StringType)
+            ))
+          }
+      }
+    }
+  }
+
+
+
+  class PravegaScan(options: CaseInsensitiveStringMap) extends Scan {
+
+    override def readSchema(): StructType = PravegaReader.pravegaSchema
+
+    private def createMetadataReader(scopeName: String,
+                                     streamName: String,
+                                     clientConfig: ClientConfig,
+                                     metadataTableName: MetadataTableName.MetadataTableName,
+                                     caseInsensitiveParams: Map[String, String]): Batch = {
+      (for (streamManager <- managed(StreamManager.create(clientConfig))) yield {
+        metadataTableName match {
+          case MetadataTableName.StreamInfo => {
+            val streamInfo = streamManager.getStreamInfo(scopeName, streamName)
+            val schema = StructType(Seq(
+              StructField("head_stream_cut", StringType),
+              StructField("tail_stream_cut", StringType)
+            ))
+            val row = new GenericInternalRow(Seq(
+              UTF8String.fromString(streamInfo.getHeadStreamCut.asText()),
+              UTF8String.fromString(streamInfo.getTailStreamCut.asText())
+            ).toArray[Any])
+            MemoryDataSourceReader(schema, Seq(row))
+          }
+          case MetadataTableName.Streams => {
+            val streams = streamManager.listStreams(scopeName)
+            val schema = StructType(Seq(
+              StructField("scope_name", StringType),
+              StructField("stream_name", StringType)
+            ))
+
+            var rows = Seq[InternalRow]()
+
+            streams.foreach(stream => {

```

Comment:
I get this warning:
```
spark-connectors/src/main/scala/io/pravega/connectors/spark/PravegaSourceProvider.scala:356: object JavaConversions in package collection is deprecated (since 2.12.0): use JavaConverters
            streams.foreach(stream => {
            ^
one warning found
```